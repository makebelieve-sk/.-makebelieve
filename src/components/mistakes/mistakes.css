.game_mistakes {
    display: flex;
    justify-content: flex-end;
}
.wrong {
    width: 35px;
    height: 49px;
    margin-right: 20px;
    background: url('./img/icon-note-inactive2.png');
    background-repeat: no-repeat;
}
.wrong-active {
    width: 35px;
    height: 49px;
    margin-right: 20px;
    background: url('./img/icon-note-wrong.png');
    background-repeat: no-repeat;
}