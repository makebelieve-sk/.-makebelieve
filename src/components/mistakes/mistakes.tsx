import React from 'react';

import './mistakes.css';

type MistakesType = {
    mistakes: number
};

export const Mistakes: React.FC<MistakesType> = ({ mistakes }) => {

    switch(mistakes) {
        case 0: 
            return (
                <div className="game_mistakes">
                    <div className="wrong"></div>
                    <div className="wrong"></div>
                    <div className="wrong"></div>
                </div>
            );
        case 1: 
            return (
                <div className="game_mistakes">
                    <div className="wrong"></div>
                    <div className="wrong"></div>
                    <div className="wrong-active"></div>
                </div>
            );
        case 2: 
            return (
                <div className="game_mistakes">
                    <div className="wrong"></div>
                    <div className="wrong-active"></div>
                    <div className="wrong-active"></div>
                </div>
            );
        case 3: 
            return (
                <div className="game_mistakes">
                    <div className="wrong-active"></div>
                    <div className="wrong-active"></div>
                    <div className="wrong-active"></div>
                </div>
            );
        default:
            return (
                <div className="game_mistakes">
                    <div className="wrong"></div>
                    <div className="wrong"></div>
                    <div className="wrong"></div>
                </div>
            );
    }
};