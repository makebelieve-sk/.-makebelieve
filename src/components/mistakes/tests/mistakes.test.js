import React from 'react';
import renderer from 'react-test-renderer';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import { Mistakes } from '../mistakes';

Enzyme.configure({ adapter: new Adapter() });

const setUp = (prop1) => shallow(
    <Mistakes mistakes={prop1} />
);

const mock = {
    mistakes: 0
};

describe(`Mistakes component`, () => {
    let component;

    beforeEach(() => {
        component = setUp(mock.mistakes);
    });

    it(`should contain div wrapper`, () => {
        const wrapper = component.find('.game_mistakes');
        expect(wrapper.length).toBe(1);
    });

    it(`Mistakes correctly renders after relaunch`, () => {
        const tree = renderer  
            .create(<Mistakes 
                mistakes={mock.mistakes}
            />)
            .toJSON();

        expect(tree).toMatchSnapshot();
    });
});