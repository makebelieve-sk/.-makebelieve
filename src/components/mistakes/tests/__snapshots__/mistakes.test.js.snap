// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Mistakes component Mistakes correctly renders after relaunch 1`] = `
<div
  className="game_mistakes"
>
  <div
    className="wrong"
  />
  <div
    className="wrong"
  />
  <div
    className="wrong"
  />
</div>
`;
